src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3247,6 +3247,9 @@ internal bool EnableEnumArrayBlockInitialization
             }
         }
 
+        internal override ImmutableArray<TypeDeclarationInfo> GetTypeDeclarationInfos(CancellationToken cancellationToken)
+            => Declarations.GetTypeDeclarationInfos(this, cancellationToken);
```

Comment:
Infos is usually not a good name.  If necessary create some sort of wrapper type aronud this, have this guy return that wrapper type.  The additional benefit there is that if we need to return *more* info in teh future, we can do that.  As opposed to now where we would be stuck since it's just an immutable array.